src/Agent.Listener/Agent.cs

```diff
@@ -326,6 +329,18 @@ private async Task<int> RunAsync(AgentSettings settings, bool runOnce = false)
             try
             {
                 Trace.Info(nameof(RunAsync));
+
+                if (AgentKnobs.CheckIfAgentIsRunningInPowershell.GetValue(HostContext).AsBoolean())
```

Comment:
Instead of attempting to iterating parent processes, can we just look for the problematic env variable or value and warn on that?

Reply:
Applied the suggestion